src/main/java/sg/ncl/MainController.java

```diff
@@ -1141,19 +1141,40 @@ public String checkJoinTeamInfo(@Valid TeamPageJoinTeamForm teamPageJoinForm, Bi
     public String experiments(Model model, HttpSession session) {
 
         List<Experiment2> experimentList = new ArrayList<>();
-        ResponseEntity responseEntity = restClient.sendGetRequest(properties.getSioExpUrl() + "/users/" + session.getAttribute("id").toString());
+        Map<Long, Realization> realizationMap = new HashMap<>();
 
-        JSONArray experimentsArray = new JSONArray(responseEntity.getBody().toString());
+        // get list of teamids
+        ResponseEntity userRespEntity = restClient.sendGetRequest(properties.getSioUsersUrl() + "/" + session.getAttribute("id"));
+
+        JSONObject object = new JSONObject(userRespEntity.getBody().toString());
+        JSONArray teamIdsJsonArray = object.getJSONArray("teams");
+
+        for (int i = 0; i < teamIdsJsonArray.length(); i++) {
+            String teamId = teamIdsJsonArray.get(i).toString();
 
-        for (int i = 0; i < experimentsArray.length(); i++) {
-            Experiment2 experiment2 = extractExperiment(experimentsArray.getJSONObject(i).toString());
-            experimentList.add(experiment2);
+            // get experiments lists of the teams
+            HttpHeaders headers = new HttpHeaders();
+            headers.setContentType(MediaType.APPLICATION_JSON);
+            headers.set("Authorization", AUTHORIZATION_HEADER);
+
+            HttpEntity<String> request = new HttpEntity<String>("parameters", headers);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Replace the type specification in this constructor call with the diamond operator ("<>"). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS2293)